### ISSUE TITLE
Question about warning messages

### ISSUE THREAD

**Issue by aistrand** [adamewing/bamsurgeon#55]
Hi Adam,

I had a couple of questions about some warning messages: what exactly does `too few reads in region (6) skipping...` mean (region (6) being an example), and what does `forced 3 reads` (3 being an example) mean? What about `could not pileup for region: 5:100015928`? Do all these error messages result in the mutation not being spiked in? If not, which ones do? Does the warning message `dropped for outcover/incover < 0.9` mean that the mutation gets dropped, or that it does not get spiked in? 

Also, you were referring to tests of bamsurgeon, but not sure exactly which ones you were talking about. In the manual it says:

`The script test/test snv.sh may be run with no parameters to display a usage statement:`
`usage: ./test_snv.sh <number of SNPs> <number of threads> <reference indexed with bwa index>`

Is this what you were referring to? Is it possible to test this out with actual data?

**Comment by adamewing** [comment]
Hi Alva,

It sounds like you're trying to spike in to a very low coverage BAM file, is that true?

The `too few reads in region` messages means there were less than a minimum number of reads covering the desired mutation site. This is controllable through a parameter `--mindepth`. The `could not pileup for region` message means where were no reads over the site you are trying to mutate. If there is insufficient coverage the mutation is not made. If a mutation is 'dropped' that means it is not spiked in.

The `forced 3 reads` is controlled by a parameter (`--minmutreads`) which defaults to 3. It ensures a minimum level of read support for mutations.

The bamsurgeon repo comes with test data, from within the `test/` folder try:

```
./test_snv.sh 1 1 ../test_data/Homo_sapiens_chr22_assembly19.fasta
```

**Comment by aistrand** [comment]
I am using a truncated BAM file, so it is possible that some of the sites have low coverage there, but the full BAM file is not low coverage (average coverage of 24). These messages only appear in the case of a few sites, so it shouldn't be too big of an issue, unless you think there is something more serious going on...

I understand how the warning messages relate to the parameters now. I set --mindepth to be 11 and --minmutreads to be 3, so that makes sense. So this means that the `too few reads in region` and `could not pileup for region` lead to a mutation not being spiked in, if I understand correctly? As long as I know which sites do not have spiked in mutations (which I do, based on the output), I should be fine.

And thanks for the information about the test data, will try that.

**Comment by adamewing** [comment]
OK, yes a few sites are probably not something to be overly worried about but you could check the coverage in some of the dropped regions e.g. with `samtools mpileup`.

Those messages indicate a mutation is not spiked in, correct. The 'snv' lines in the log files and/or the output of `makevcf.py` will only contain sites where a spikein actually occured.

**Comment by aistrand** [comment]
OK, makes sense.

I'm not sure what you mean by "snv" lines, my log files only have lines of the type: `read HWI-ST1329:287:C2JDBACXX:3:2312:18024:34558,105000066,F TTAAGTATTAAATAGGGCTGTAGGAATAAGAAACTAAAAACAAAATGGATTATTCTGTAAAGAAATTTGGACCCTACTGAATAGTGTAAAGTATCCTTTC`

I do get output from makevcf.py, I believe, I get output from standard output, and that's where I get information about what is going on with the program.

**Comment by aistrand** [comment]
I tried to run a newly created bam file (with spiked in mutations) as input to a program I am using, but it seems that I have to sort the bam file as I got the following message:

`[bam_pileup_core] the input is not sorted (reads out of order)`
`[bam_plp_destroy] memory leak: 33. Continue anyway.`
`[bam_plp_destroy] memory leak: 35. Continue anyway.`
`[bam_plp_destroy] memory leak: 18. Continue anyway.`

Why are the reads out of order with bamsurgeon?
